### PR TITLE
 Fix riot-web Promise.defer warnings (#7409)

### PR DESCRIPTION
**Pull request by aaronraimist**
More progress on #7409

### CODE REVIEW

File: src/vector/platform/WebPlatform.js [hunk 1]
```diff
@@ -103,31 +103,31 @@ export default class WebPlatform extends VectorBasePlatform {
...
+                        if (err === null) err = { status: response.status };
+                        reject(err);
+                        return;
+                    }
+
+                    const ver = body.trim();
+                    resolve(ver);
+                },
+            );
+        })
```
Comment (bwindels) [inline src/vector/platform/WebPlatform.js]: Nit: missing semicolon

File: src/vector/platform/WebPlatform.js [hunk 2]
```diff
@@ -68,11 +68,11 @@ export default class WebPlatform extends VectorBasePlatform {
...
-        const defer = Promise.defer();
-        global.Notification.requestPermission((result) => {
-            defer.resolve(result);
-        });
-        return defer.promise;
+        return new Promise(function(resolve, reject) {
+            global.Notification.requestPermission((result) => {
+                resolve(result);
+            });
+        })
```
Comment (bwindels) [inline src/vector/platform/WebPlatform.js]: Nit: missing semicolon

File: src/vector/index.js [hunk 3]
```diff
@@ -175,37 +175,35 @@ function makeRegistrationUrl(params) {
...
 
-    return deferred.promise;
+                // We parse the JSON ourselves rather than use the JSON
+                // parameter, since this throws a parse error on empty
+                // which breaks if there's no config.json and we're
+                // loading from the filesystem (see above).
+                resolve(JSON.parse(body));
+            }
+        );
+    })
```
Comment (bwindels) [inline src/vector/index.js]: Nit: missing semicolon

File: src/vector/getconfig.js [hunk 4]
```diff
@@ -33,35 +33,33 @@ export async function getVectorConfig(relativeLocation) {
...
-
-    return deferred.promise;
+                // We parse the JSON ourselves rather than use the JSON
+                // parameter, since this throws a parse error on empty
+                // which breaks if there's no config.json and we're
+                // loading from the filesystem (see above).
+                resolve(JSON.parse(body));
+            }
+        );
+    })
```
Comment (bwindels) [inline src/vector/getconfig.js]: Nit: missing semicolon

**Comment by dbkr** [comment]
(DCO signoff is in the commit messages)

**Comment by aaronraimist** [comment]
This should be ready to go @bwindels

**Comment by turt2live** [comment]
Thank you!

**Pull request merged**